12-image-gen/src/entry.py

```diff
@@ -0,0 +1,349 @@
+        - width: Image width (default: 800)
+        - height: Image height (default: 400)
+        - color1: Start color in hex (default: random)
+        - color2: End color in hex (default: random)
+        """
+        # Get dimensions from query params or use defaults
+        width = int(params.get("width", [800])[0])
+        height = int(params.get("height", [400])[0])
+
+        # Get colors or generate random ones
+        color1 = params.get("color1", [None])[0]
+        color2 = params.get("color2", [None])[0]
+
+        if not color1:
+            color1 = "#{:06x}".format(random.randint(0, 0xFFFFFF))
+        if not color2:
+            color2 = "#{:06x}".format(random.randint(0, 0xFFFFFF))
+
+        # Convert hex colors to RGB tuples
+        r1, g1, b1 = self.hex_to_rgb(color1)
+        r2, g2, b2 = self.hex_to_rgb(color2)
+
+        # Create a new image with RGB mode
+        image = Image.new("RGB", (width, height))
+        draw = ImageDraw.Draw(image)
+
+        # Draw gradient by interpolating between colors
+        for y in range(height):
+            # Calculate interpolation factor (0.0 to 1.0)
+            factor = y / height
+
+            # Interpolate each color channel
+            r = int(r1 + (r2 - r1) * factor)
+            g = int(g1 + (g2 - g1) * factor)
+            b = int(b1 + (b2 - b1) * factor)
+
+            # Draw a horizontal line with the interpolated color
+            draw.line([(0, y), (width, y)], fill=(r, g, b))
+
+        # Convert image to bytes and return as PNG
+        return self.image_to_response(image, "image/png")
+
+    def generate_badge(self, params: dict) -> Response:
+        """
+        Generate a badge/button with custom text.
+
+        Query parameters:
+        - text: Badge text (default: "Hello World")
+        - bg_color: Background color in hex (default: #4CAF50)
+        - text_color: Text color in hex (default: #FFFFFF)
+        """
+        # Get parameters
+        text = params.get("text", ["Hello World"])[0]
+        bg_color = params.get("bg_color", ["#4CAF50"])[0]
+        text_color = params.get("text_color", ["#FFFFFF"])[0]
+
+        # Convert hex colors to RGB
+        bg_rgb = self.hex_to_rgb(bg_color)
+        text_rgb = self.hex_to_rgb(text_color)
+
+        # Create image with padding for text
+        # We'll estimate size based on text length
+        padding = 20
+        font_size = 32
+
+        # Create a temporary image to measure text size
+        temp_img = Image.new("RGB", (1, 1))
+        temp_draw = ImageDraw.Draw(temp_img)
+
+        # Use default font (Pillow's built-in font)
+        # Note: In a production environment, you might want to include custom fonts
+        try:
+            # Try to use a better font if available
+            font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size)
+        except Exception:
+            # Fall back to default font
+            font = ImageFont.load_default()
```

Comment:
```suggestion
            font = ImageFont.load_default()
```

We don't have local file system access don't we?